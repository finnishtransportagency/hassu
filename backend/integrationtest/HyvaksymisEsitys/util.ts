import { config } from "../../src/config";
import { getS3Client } from "../../src/aws/client";
import { DeleteObjectsCommand, ListObjectsV2Command, ListObjectsV2CommandOutput, PutObjectCommand } from "@aws-sdk/client-s3";
import { createPresignedPost } from "@aws-sdk/s3-presigned-post";
import { adaptFileName } from "../../src/tiedostot/paths";
import { getAxios } from "../../src/aws/monitoring";
import FormData from "form-data";

export async function insertYllapitoFileToS3(pathInS3: string) {
  if (pathInS3.includes(config.yllapitoBucketName)) {
    throw new Error("path ei saa sisältää ylläpitoBucketNamea");
  }
  const buffer = Buffer.from("Test");
  await getS3Client().send(
    new PutObjectCommand({
      Body: buffer,
      Bucket: config.yllapitoBucketName,
      Key: pathInS3,
    })
  );
}

export async function insertUploadFileToS3(uuid: string, filename: string) {
  const fileNameWithPath = `${uuid}/${adaptFileName(filename)}`;
  const presignedPost = await createPresignedPost(getS3Client(), {
    Bucket: config.uploadBucketName,
    Key: fileNameWithPath,
    Expires: 600,
  });
  const fields = presignedPost.fields;
  const form = new FormData();
  Object.keys(fields).forEach((key) => {
    form.append(key, fields[key]);
  });
  form.append("file", "Test");
  const axios = getAxios();
  const lengthSync = form.getLengthSync();

  await axios.post(presignedPost.url, form, {
    headers: {
      "Content-Length": lengthSync,
      ...form.getHeaders(),
    },
  });
}

export async function deleteYllapitoFiles(sourcePrefix?: string) {
  if (sourcePrefix && sourcePrefix.includes(config.yllapitoBucketName)) {
    throw new Error("sourcePrefix ei saa sisältää yllapitoBucketNamea");
  }
  const s3 = getS3Client();
  let ContinuationToken = undefined;
  do {
    const { Contents = [], NextContinuationToken }: ListObjectsV2CommandOutput = await s3.send(
      new ListObjectsV2Command({
        Bucket: config.yllapitoBucketName,
        Prefix: sourcePrefix,
        ContinuationToken,
      })
    );
    await deleteFiles(Contents.map(({ Key }) => Key as string));

    ContinuationToken = NextContinuationToken;
  } while (ContinuationToken);
}

export async function emptyUploadFiles() {
  const s3 = getS3Client();
  let ContinuationToken = undefined;
  do {
    const { Contents = [], NextContinuationToken }: ListObjectsV2CommandOutput = await s3.send(
      new ListObjectsV2Command({
        Bucket: config.uploadBucketName,
        ContinuationToken,
      })
    );
    await deleteFiles(
      Contents.map(({ Key }) => Key as string),
      config.uploadBucketName
    );

    ContinuationToken = NextContinuationToken;
  } while (ContinuationToken);
}

async function deleteFiles(paths: string[], bucket: string = config.yllapitoBucketName) {
  if (paths.some((path) => path.includes(config.yllapitoBucketName))) {
    throw new Error("path ei saa sisältää ylläpitoBucketNamea");
  }
  await Promise.all(
    // DeleteObjectsCommand takes at most 1000 items
    getChunksOfThousand(paths).map((paths) =>
      getS3Client().send(
        new DeleteObjectsCommand({
          Bucket: bucket,
          Delete: { Objects: paths.map((path) => ({ Key: path })) },
        })
      )
    )
  );
}

function getChunksOfThousand<T>(array: T[]): T[][] {
  const chunks: T[][] = [];
  for (let i = 0; i < array.length; i += 1000) {
    chunks.push(array.slice(i, i + 1000));
  }
  return chunks;
}

export async function getYllapitoFilesUnderPath(pathInS3: string): Promise<string[]> {
  const s3 = getS3Client();
  const files: string[] = [];
  let ContinuationToken = undefined;
  do {
    const { Contents = [], NextContinuationToken }: ListObjectsV2CommandOutput = await s3.send(
      new ListObjectsV2Command({
        Bucket: config.yllapitoBucketName,
        Prefix: pathInS3,
        ContinuationToken,
      })
    );
    files.push(...Contents.map(({ Key }) => Key as string));
    ContinuationToken = NextContinuationToken;
  } while (ContinuationToken);
  return files;
}
