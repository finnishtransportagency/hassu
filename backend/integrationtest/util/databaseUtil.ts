import * as log from "loglevel";

import * as sinon from "sinon";
import { SinonStub } from "sinon";
import * as dynamoDB from "../../src/database/dynamoDB";

import DynamoDB from "aws-sdk/clients/dynamodb";
import { HookFunction } from "mocha";

const localDynamoDBParams = {
  endpoint: "http://localhost:4566",
  accessKeyId: "anyKey",
  secretAccessKey: "anySecret",
  region: "eu-west-1",
};
const localDynamoDB = new DynamoDB(localDynamoDBParams);
const localDocumentClient = new DynamoDB.DocumentClient({
  service: localDynamoDB,
  apiVersion: "2012-08-10",
  params: localDynamoDBParams,
});

// Try to use the database configuration from deployment in
let localDynamoDBDocumentClientStub: SinonStub;

function replaceAWSDynamoDBWithLocalstack() {
  localDynamoDBDocumentClientStub = sinon.stub(dynamoDB, "getDynamoDBDocumentClient");
  localDynamoDBDocumentClientStub.returns(localDocumentClient);
}

export async function setupLocalDatabase() {
  try {
    await deleteAllItemsFromDatabase();
  } catch (e) {
    // Ignore
  }
  replaceAWSDynamoDBWithLocalstack();
}

async function deleteAllItemsFromDatabase() {
  // Hard-code table name to prevent accidental deletion from AWS
  log.info("Cleaning up database");
  const items = (await localDocumentClient.scan({ TableName: "Projekti-localstack" }).promise()).Items;
  if (items) {
    await Promise.all(
      items.map(async (item) => {
        log.info("Deleting ", item);
        await localDocumentClient.delete({ TableName: "Projekti-localstack", Key: { oid: item.oid } }).promise();
      })
    );
  }
}

afterEach("Reset database stub", (() => {
  return localDynamoDBDocumentClientStub.restore();
}) as HookFunction);
