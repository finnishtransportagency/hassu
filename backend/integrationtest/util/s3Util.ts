import { produceAWSClient } from "../../src/aws/clientProducer";
import { S3Client } from "@aws-sdk/client-s3";

export function localstackS3Client() {
  produceAWSClient(
    "s3",
    () =>
      new S3Client({
        endpoint: "http://localhost:4566",
        forcePathStyle: true,
      }),
    true
  );
}
