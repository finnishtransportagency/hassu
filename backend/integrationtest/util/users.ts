import * as userService from "../../src/service/userService";
import * as sinon from "sinon";
import { UserFixture } from "../../test/fixture/userFixture";

export function runAsVaylaUser() {
  sinon.stub(userService, "identifyUser").resolves();
  userService.identifyMockUser(UserFixture.mattiMeikalainen);
}

export function runAsAnonymous() {
  userService.identifyMockUser();
}
