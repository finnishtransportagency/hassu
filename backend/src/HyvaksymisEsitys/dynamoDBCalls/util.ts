import { UpdateCommand } from "@aws-sdk/lib-dynamodb";
import { log } from "../../logger";
import { getDynamoDBDocumentClient } from "../../aws/client";
import { ConditionalCheckFailedException } from "@aws-sdk/client-dynamodb";
import { SimultaneousUpdateError } from "hassu-common/error";

export async function sendUpdateCommandToDynamoDB(params: UpdateCommand): Promise<void> {
  if (log.isLevelEnabled("debug")) {
    log.debug("Updating projekti to Hassu with params", { params });
  }
  try {
    const dynamoDBDocumentClient = getDynamoDBDocumentClient();
    await dynamoDBDocumentClient.send(params);
  } catch (e) {
    if (e instanceof ConditionalCheckFailedException) {
      throw new SimultaneousUpdateError("Projektia on p√§ivitetty tietokannassa. Lataa projekti uudelleen.");
    }
    log.error(e instanceof Error ? e.message : String(e), { params });
    throw e;
  }
}
