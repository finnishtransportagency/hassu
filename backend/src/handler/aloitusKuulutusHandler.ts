import { AloitusKuulutusTila, TilaSiirtymaInput, TilasiirtymaToiminto } from "../../../common/graphql/apiModel";
import { requirePermissionLuku, requirePermissionMuokkaa } from "../user";
import { requireProjektiPaallikko } from "../user/userService";
import { projektiDatabase } from "../database/projektiDatabase";
import { asiakirjaAdapter } from "./asiakirjaAdapter";
import { AloitusKuulutus, AloitusKuulutusJulkaisu, DBProjekti } from "../database/model/projekti";

function findAloitusKuulutusWaitingForApproval(projekti: DBProjekti): AloitusKuulutusJulkaisu | undefined {
  if (projekti.aloitusKuulutusJulkaisut) {
    return projekti.aloitusKuulutusJulkaisut
      .filter((julkaisu) => julkaisu.tila == AloitusKuulutusTila.ODOTTAA_HYVAKSYNTAA)
      .pop();
  }
}

async function sendForApproval(projekti: DBProjekti, aloitusKuulutus: AloitusKuulutus) {
  const muokkaaja = requirePermissionMuokkaa(projekti);
  const julkaisuWaitingForApproval = findAloitusKuulutusWaitingForApproval(projekti);
  if (julkaisuWaitingForApproval) {
    throw new Error("Aloituskuulutus on jo olemassa odottamassa hyväksyntää");
  }

  await removeRejectionReasonIfExists(projekti, aloitusKuulutus);

  const aloitusKuulutusJulkaisu = asiakirjaAdapter.adaptAloitusKuulutusJulkaisu(projekti);
  aloitusKuulutusJulkaisu.tila = AloitusKuulutusTila.ODOTTAA_HYVAKSYNTAA;
  aloitusKuulutusJulkaisu.muokkaaja = muokkaaja.uid;
  await projektiDatabase.insertAloitusKuulutusJulkaisu(projekti.oid, aloitusKuulutusJulkaisu);
}

async function reject(projekti: DBProjekti, aloitusKuulutus: AloitusKuulutus, syy: string) {
  requireProjektiPaallikko(projekti);
  const julkaisuWaitingForApproval = findAloitusKuulutusWaitingForApproval(projekti);
  if (!julkaisuWaitingForApproval) {
    throw new Error("Ei aloituskuulutusta odottamassa hyväksyntää");
  }
  aloitusKuulutus.palautusSyy = syy;
  await projektiDatabase.saveProjekti({ oid: projekti.oid, aloitusKuulutus });
  await projektiDatabase.deleteAloitusKuulutusJulkaisu(projekti, julkaisuWaitingForApproval);
}

async function approve(projekti: DBProjekti, aloitusKuulutus: AloitusKuulutus) {
  const projektiPaallikko = requireProjektiPaallikko(projekti);
  const julkaisuWaitingForApproval = findAloitusKuulutusWaitingForApproval(projekti);
  if (!julkaisuWaitingForApproval) {
    throw new Error("Ei aloituskuulutusta odottamassa hyväksyntää");
  }
  await removeRejectionReasonIfExists(projekti, aloitusKuulutus);
  julkaisuWaitingForApproval.tila = AloitusKuulutusTila.HYVAKSYTTY;
  julkaisuWaitingForApproval.hyvaksyja = projektiPaallikko.uid;
  await projektiDatabase.updateAloitusKuulutusJulkaisu(projekti, julkaisuWaitingForApproval);
}

async function removeRejectionReasonIfExists(projekti: DBProjekti, aloitusKuulutus: AloitusKuulutus) {
  if (aloitusKuulutus.palautusSyy) {
    aloitusKuulutus.palautusSyy = null;
    await projektiDatabase.saveProjekti({ oid: projekti.oid, aloitusKuulutus });
  }
}

class AloitusKuulutusHandler {
  async siirraTila({ oid, syy, toiminto }: TilaSiirtymaInput) {
    requirePermissionLuku();
    const projekti = await projektiDatabase.loadProjektiByOid(oid);
    const aloitusKuulutus = projekti.aloitusKuulutus;
    if (!aloitusKuulutus) {
      throw new Error("Projektilla ei ole aloituskuulutusta");
    }

    if (toiminto == TilasiirtymaToiminto.LAHETA_HYVAKSYTTAVAKSI) {
      await sendForApproval(projekti, aloitusKuulutus);
    } else if (toiminto == TilasiirtymaToiminto.HYLKAA) {
      await reject(projekti, aloitusKuulutus, syy);
    } else if (toiminto == TilasiirtymaToiminto.HYVAKSY) {
      await approve(projekti, aloitusKuulutus);
    } else {
      throw new Error("Tuntematon toiminto");
    }

    return Promise.resolve(undefined);
  }
}

export const aloitusKuulutusHandler = new AloitusKuulutusHandler();
