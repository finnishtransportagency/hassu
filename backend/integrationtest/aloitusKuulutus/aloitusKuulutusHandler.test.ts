import { describe, it } from "mocha";
import { setupLocalDatabase } from "../util/databaseUtil";
import * as sinon from "sinon";
import { projektiDatabase } from "../../src/database/projektiDatabase";
import { ProjektiFixture } from "../../test/fixture/projektiFixture";
import { TilasiirtymaToiminto } from "../../../common/graphql/apiModel";
import { aloitusKuulutusHandler } from "../../src/handler/aloitusKuulutusHandler";
import { UserFixture } from "../../test/fixture/userFixture";
import { userService } from "../../src/user";

const { expect } = require("chai");

async function takeSnapshot(oid: string) {
  const dbProjekti = await projektiDatabase.loadProjektiByOid(oid);
  expect({
    aloitusKuulutus: dbProjekti.aloitusKuulutus,
    aloitusKuulutusJulkaisut: dbProjekti.aloitusKuulutusJulkaisut,
  }).toMatchSnapshot();
}

describe("AloitusKuulutus", () => {
  let userFixture: UserFixture;

  afterEach(() => {
    userFixture.logout();
    sinon.reset();
    sinon.restore();
  });

  beforeEach("Initialize test database!", async () => await setupLocalDatabase());

  beforeEach(() => {
    userFixture = new UserFixture(userService);
  });

  it("should create and manipulate projekti successfully", async function () {
    userFixture.loginAs(UserFixture.mattiMeikalainen);
    const projekti = new ProjektiFixture().dbProjekti1;
    const oid = projekti.oid;
    await projektiDatabase.createProjekti(projekti);
    await takeSnapshot(oid);

    await aloitusKuulutusHandler.siirraTila({ oid, toiminto: TilasiirtymaToiminto.LAHETA_HYVAKSYTTAVAKSI });
    await takeSnapshot(oid);

    userFixture.loginAs(UserFixture.pekkaProjari);
    await aloitusKuulutusHandler.siirraTila({ oid, toiminto: TilasiirtymaToiminto.HYLKAA, syy: "Korjaa teksti" });
    await takeSnapshot(oid);

    userFixture.loginAs(UserFixture.mattiMeikalainen);
    await aloitusKuulutusHandler.siirraTila({ oid, toiminto: TilasiirtymaToiminto.LAHETA_HYVAKSYTTAVAKSI });
    await takeSnapshot(oid);

    userFixture.loginAs(UserFixture.pekkaProjari);
    await aloitusKuulutusHandler.siirraTila({ oid, toiminto: TilasiirtymaToiminto.HYVAKSY });
    await takeSnapshot(oid);
  });
});
