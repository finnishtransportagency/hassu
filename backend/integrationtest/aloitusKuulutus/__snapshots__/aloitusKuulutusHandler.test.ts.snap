// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AloitusKuulutus should create and manipulate projekti successfully 1`] = `
Object {
  "aloitusKuulutus": Object {
    "hankkeenKuvaus": Object {
      "RUOTSI": "På svenska",
      "SAAME": "Saameksi",
      "SUOMI": "Lorem Ipsum",
    },
    "id": 1,
    "kuulutusPaiva": "2022-01-02",
    "kuulutusYhteystiedot": Object {
      "yhteysHenkilot": Array [
        "A000124",
      ],
      "yhteysTiedot": Array [
        Object {
          "etunimi": "Marko",
          "organisaatio": "Kajaani",
          "puhelinnumero": "0293121213",
          "sahkoposti": "markku.koi@koi.com",
          "sukunimi": "Koi",
        },
      ],
    },
    "siirtyySuunnitteluVaiheeseen": "2022-01-01",
  },
  "aloitusKuulutusJulkaisut": undefined,
}
`;

exports[`AloitusKuulutus should create and manipulate projekti successfully 2`] = `
Object {
  "aloitusKuulutus": Object {
    "hankkeenKuvaus": Object {
      "RUOTSI": "På svenska",
      "SAAME": "Saameksi",
      "SUOMI": "Lorem Ipsum",
    },
    "id": 1,
    "kuulutusPaiva": "2022-01-02",
    "kuulutusYhteystiedot": Object {
      "yhteysHenkilot": Array [
        "A000124",
      ],
      "yhteysTiedot": Array [
        Object {
          "etunimi": "Marko",
          "organisaatio": "Kajaani",
          "puhelinnumero": "0293121213",
          "sahkoposti": "markku.koi@koi.com",
          "sukunimi": "Koi",
        },
      ],
    },
    "siirtyySuunnitteluVaiheeseen": "2022-01-01",
  },
  "aloitusKuulutusJulkaisut": Array [
    Object {
      "aloituskuulutusPDFt": Object {
        "RUOTSI": Object {
          "aloituskuulutusIlmoitusPDFPath": "/aloituskuulutus/1/T412_1 Meddelande om kungorelse om inledande av planering.pdf",
          "aloituskuulutusPDFPath": "/aloituskuulutus/1/T412 Kungorelse om inledande av planering.pdf",
        },
        "SUOMI": Object {
          "aloituskuulutusIlmoitusPDFPath": "/aloituskuulutus/1/T412_1 Ilmoitus aloituskuulutuksesta.pdf",
          "aloituskuulutusPDFPath": "/aloituskuulutus/1/T412 Aloituskuulutus.pdf",
        },
      },
      "hankkeenKuvaus": Object {
        "RUOTSI": "På svenska",
        "SAAME": "Saameksi",
        "SUOMI": "Lorem Ipsum",
      },
      "id": 1,
      "kielitiedot": Object {
        "ensisijainenKieli": "SUOMI",
        "projektinNimiVieraskielella": "Namnet på svenska",
        "toissijainenKieli": "RUOTSI",
      },
      "kuulutusPaiva": "2022-01-02",
      "muokkaaja": "A000111",
      "siirtyySuunnitteluVaiheeseen": "2022-01-01",
      "suunnitteluSopimus": Object {
        "__typename": "SuunnitteluSopimusJulkaisu",
        "email": "Kunta.Kuntalainen@vayla.fi",
        "etunimi": "Kunta",
        "kunta": 536,
        "logo": "/suunnittelusopimus/logo.png",
        "puhelinnumero": "123456789",
        "sukunimi": "Kuntalainen",
      },
      "tila": "ODOTTAA_HYVAKSYNTAA",
      "velho": Object {
        "asiatunnusELY": "ELY/1/2022",
        "asiatunnusVayla": "VAYLA/1/2022",
        "kunnat": Array [
          837,
          536,
        ],
        "maakunnat": Array [
          1,
          6,
        ],
        "nimi": "Testiprojekti 1",
        "suunnittelustaVastaavaViranomainen": "UUDENMAAN_ELY",
        "tyyppi": "TIE",
        "vaylamuoto": Array [
          "tie",
        ],
      },
      "yhteystiedot": Array [
        Object {
          "etunimi": "Kunta",
          "kunta": 536,
          "organisaatio": "Nokia",
          "puhelinnumero": "123456789",
          "sahkoposti": "Kunta.Kuntalainen@vayla.fi",
          "sukunimi": "Kuntalainen",
        },
        Object {
          "elyOrganisaatio": "PIRKANMAAN_ELY",
          "etunimi": "Eemil",
          "organisaatio": "ELY",
          "puhelinnumero": "123456789",
          "sahkoposti": "eemil.elylainen@ely.fi",
          "sukunimi": "Elylainen",
        },
        Object {
          "etunimi": "Marko",
          "organisaatio": "Kajaani",
          "puhelinnumero": "0293121213",
          "sahkoposti": "markku.koi@koi.com",
          "sukunimi": "Koi",
        },
      ],
    },
  ],
}
`;

exports[`AloitusKuulutus should create and manipulate projekti successfully 3`] = `
Object {
  "aloitusKuulutus": Object {
    "hankkeenKuvaus": Object {
      "RUOTSI": "På svenska",
      "SAAME": "Saameksi",
      "SUOMI": "Lorem Ipsum",
    },
    "id": 1,
    "kuulutusPaiva": "2022-01-02",
    "kuulutusYhteystiedot": Object {
      "yhteysHenkilot": Array [
        "A000124",
      ],
      "yhteysTiedot": Array [
        Object {
          "etunimi": "Marko",
          "organisaatio": "Kajaani",
          "puhelinnumero": "0293121213",
          "sahkoposti": "markku.koi@koi.com",
          "sukunimi": "Koi",
        },
      ],
    },
    "palautusSyy": "Korjaa teksti",
    "siirtyySuunnitteluVaiheeseen": "2022-01-01",
  },
  "aloitusKuulutusJulkaisut": Array [],
}
`;

exports[`AloitusKuulutus should create and manipulate projekti successfully 4`] = `
Object {
  "aloitusKuulutus": Object {
    "hankkeenKuvaus": Object {
      "RUOTSI": "På svenska",
      "SAAME": "Saameksi",
      "SUOMI": "Lorem Ipsum",
    },
    "id": 1,
    "kuulutusPaiva": "2022-01-02",
    "kuulutusYhteystiedot": Object {
      "yhteysHenkilot": Array [
        "A000124",
      ],
      "yhteysTiedot": Array [
        Object {
          "etunimi": "Marko",
          "organisaatio": "Kajaani",
          "puhelinnumero": "0293121213",
          "sahkoposti": "markku.koi@koi.com",
          "sukunimi": "Koi",
        },
      ],
    },
    "palautusSyy": null,
    "siirtyySuunnitteluVaiheeseen": "2022-01-01",
  },
  "aloitusKuulutusJulkaisut": Array [
    Object {
      "aloituskuulutusPDFt": Object {
        "RUOTSI": Object {
          "aloituskuulutusIlmoitusPDFPath": "/aloituskuulutus/1/T412_1 Meddelande om kungorelse om inledande av planering.pdf",
          "aloituskuulutusPDFPath": "/aloituskuulutus/1/T412 Kungorelse om inledande av planering.pdf",
        },
        "SUOMI": Object {
          "aloituskuulutusIlmoitusPDFPath": "/aloituskuulutus/1/T412_1 Ilmoitus aloituskuulutuksesta.pdf",
          "aloituskuulutusPDFPath": "/aloituskuulutus/1/T412 Aloituskuulutus.pdf",
        },
      },
      "hankkeenKuvaus": Object {
        "RUOTSI": "På svenska",
        "SAAME": "Saameksi",
        "SUOMI": "Lorem Ipsum",
      },
      "id": 1,
      "kielitiedot": Object {
        "ensisijainenKieli": "SUOMI",
        "projektinNimiVieraskielella": "Namnet på svenska",
        "toissijainenKieli": "RUOTSI",
      },
      "kuulutusPaiva": "2022-01-02",
      "muokkaaja": "A000111",
      "siirtyySuunnitteluVaiheeseen": "2022-01-01",
      "suunnitteluSopimus": Object {
        "__typename": "SuunnitteluSopimusJulkaisu",
        "email": "Kunta.Kuntalainen@vayla.fi",
        "etunimi": "Kunta",
        "kunta": 536,
        "logo": "/suunnittelusopimus/logo.png",
        "puhelinnumero": "123456789",
        "sukunimi": "Kuntalainen",
      },
      "tila": "ODOTTAA_HYVAKSYNTAA",
      "velho": Object {
        "asiatunnusELY": "ELY/1/2022",
        "asiatunnusVayla": "VAYLA/1/2022",
        "kunnat": Array [
          837,
          536,
        ],
        "maakunnat": Array [
          1,
          6,
        ],
        "nimi": "Testiprojekti 1",
        "suunnittelustaVastaavaViranomainen": "UUDENMAAN_ELY",
        "tyyppi": "TIE",
        "vaylamuoto": Array [
          "tie",
        ],
      },
      "yhteystiedot": Array [
        Object {
          "etunimi": "Kunta",
          "kunta": 536,
          "organisaatio": "Nokia",
          "puhelinnumero": "123456789",
          "sahkoposti": "Kunta.Kuntalainen@vayla.fi",
          "sukunimi": "Kuntalainen",
        },
        Object {
          "elyOrganisaatio": "PIRKANMAAN_ELY",
          "etunimi": "Eemil",
          "organisaatio": "ELY",
          "puhelinnumero": "123456789",
          "sahkoposti": "eemil.elylainen@ely.fi",
          "sukunimi": "Elylainen",
        },
        Object {
          "etunimi": "Marko",
          "organisaatio": "Kajaani",
          "puhelinnumero": "0293121213",
          "sahkoposti": "markku.koi@koi.com",
          "sukunimi": "Koi",
        },
      ],
    },
  ],
}
`;

exports[`AloitusKuulutus should create and manipulate projekti successfully 5`] = `
Object {
  "aloitusKuulutus": Object {
    "hankkeenKuvaus": Object {
      "RUOTSI": "På svenska",
      "SAAME": "Saameksi",
      "SUOMI": "Lorem Ipsum",
    },
    "id": 1,
    "kuulutusPaiva": "2022-01-02",
    "kuulutusYhteystiedot": Object {
      "yhteysHenkilot": Array [
        "A000124",
      ],
      "yhteysTiedot": Array [
        Object {
          "etunimi": "Marko",
          "organisaatio": "Kajaani",
          "puhelinnumero": "0293121213",
          "sahkoposti": "markku.koi@koi.com",
          "sukunimi": "Koi",
        },
      ],
    },
    "palautusSyy": null,
    "siirtyySuunnitteluVaiheeseen": "2022-01-01",
  },
  "aloitusKuulutusJulkaisut": Array [
    Object {
      "aloituskuulutusPDFt": Object {
        "RUOTSI": Object {
          "aloituskuulutusIlmoitusPDFPath": "/aloituskuulutus/1/T412_1 Meddelande om kungorelse om inledande av planering.pdf",
          "aloituskuulutusPDFPath": "/aloituskuulutus/1/T412 Kungorelse om inledande av planering.pdf",
        },
        "SUOMI": Object {
          "aloituskuulutusIlmoitusPDFPath": "/aloituskuulutus/1/T412_1 Ilmoitus aloituskuulutuksesta.pdf",
          "aloituskuulutusPDFPath": "/aloituskuulutus/1/T412 Aloituskuulutus.pdf",
        },
      },
      "hankkeenKuvaus": Object {
        "RUOTSI": "På svenska",
        "SAAME": "Saameksi",
        "SUOMI": "Lorem Ipsum",
      },
      "hyvaksyja": "A123",
      "hyvaksymisPaiva": "2022-11-04",
      "id": 1,
      "kielitiedot": Object {
        "ensisijainenKieli": "SUOMI",
        "projektinNimiVieraskielella": "Namnet på svenska",
        "toissijainenKieli": "RUOTSI",
      },
      "kuulutusPaiva": "2022-01-02",
      "muokkaaja": "A000111",
      "siirtyySuunnitteluVaiheeseen": "2022-01-01",
      "suunnitteluSopimus": Object {
        "__typename": "SuunnitteluSopimusJulkaisu",
        "email": "Kunta.Kuntalainen@vayla.fi",
        "etunimi": "Kunta",
        "kunta": 536,
        "logo": "/suunnittelusopimus/logo.png",
        "puhelinnumero": "123456789",
        "sukunimi": "Kuntalainen",
      },
      "tila": "HYVAKSYTTY",
      "velho": Object {
        "asiatunnusELY": "ELY/1/2022",
        "asiatunnusVayla": "VAYLA/1/2022",
        "kunnat": Array [
          837,
          536,
        ],
        "maakunnat": Array [
          1,
          6,
        ],
        "nimi": "Testiprojekti 1",
        "suunnittelustaVastaavaViranomainen": "UUDENMAAN_ELY",
        "tyyppi": "TIE",
        "vaylamuoto": Array [
          "tie",
        ],
      },
      "yhteystiedot": Array [
        Object {
          "etunimi": "Kunta",
          "kunta": 536,
          "organisaatio": "Nokia",
          "puhelinnumero": "123456789",
          "sahkoposti": "Kunta.Kuntalainen@vayla.fi",
          "sukunimi": "Kuntalainen",
        },
        Object {
          "elyOrganisaatio": "PIRKANMAAN_ELY",
          "etunimi": "Eemil",
          "organisaatio": "ELY",
          "puhelinnumero": "123456789",
          "sahkoposti": "eemil.elylainen@ely.fi",
          "sukunimi": "Elylainen",
        },
        Object {
          "etunimi": "Marko",
          "organisaatio": "Kajaani",
          "puhelinnumero": "0293121213",
          "sahkoposti": "markku.koi@koi.com",
          "sukunimi": "Koi",
        },
      ],
    },
  ],
}
`;

exports[`AloitusKuulutus should create and manipulate projekti successfully 6`] = `
Object {
  "args": Object {
    "DistributionId": "unit-test-distribution-id",
    "InvalidationBatch": Object {
      "CallerReference": "***unittest***",
      "Paths": Object {
        "Items": Array [
          "/tiedostot/suunnitelma/1/aloituskuulutus/*",
        ],
        "Quantity": 1,
      },
    },
  },
  "stub": "createInvalidation",
}
`;

exports[`AloitusKuulutus should create and manipulate projekti successfully 7`] = `
Array [
  Object {
    "subject": "Valtion liikenneväylien suunnittelu: Aloituskuulutus odottaa hyväksyntää ELY/1/2022",
    "text": "Valtion liikenneväylien suunnittelu -järjestelmän projektistasi
Testiprojekti 1
on luotu aloituskuulutus, joka odottaa hyväksyntääsi.
Voit tarkastella projektia osoitteessa https://localhost:3000/yllapito/projekti/1
Saat tämän viestin, koska sinut on merkitty projektin projektipäälliköksi. Tämä on automaattinen sähköposti, johon ei voi vastata.",
    "to": Array [
      "pekka.projari@vayla.fi",
    ],
  },
  Object {
    "subject": "Valtion liikenneväylien suunnittelu: Aloituskuulutus odottaa hyväksyntää ELY/1/2022",
    "text": "Valtion liikenneväylien suunnittelu -järjestelmän projektistasi
Testiprojekti 1
on luotu aloituskuulutus, joka odottaa hyväksyntääsi.
Voit tarkastella projektia osoitteessa https://localhost:3000/yllapito/projekti/1
Saat tämän viestin, koska sinut on merkitty projektin projektipäälliköksi. Tämä on automaattinen sähköposti, johon ei voi vastata.",
    "to": Array [
      "pekka.projari@vayla.fi",
    ],
  },
  Object {
    "subject": "Valtion liikenneväylien suunnittelu: Aloituskuulutus hyväksytty ELY/1/2022",
    "text": "Valtion liikenneväylien suunnittelu -järjestelmän projektin
Testiprojekti 1
aloituskuulutus on hyväksytty.
Voit tarkastella aloituskuulutusta osoitteessa https://localhost:3000/yllapito/projekti/1/aloituskuulutus
Saat tämän viestin, koska sinut on merkitty aloituskuulutuksen laatijaksi. Tämä on automaattinen sähköposti, johon ei voi vastata.",
    "to": "mikko.haapamaki@cgi.com",
  },
  Object {
    "attachments": Array [
      Object {
        "contentType": "application/pdf",
        "filename": "T412 Aloituskuulutus.pdf",
      },
    ],
    "subject": "Valtion liikenneväylien suunnittelu: Aloituskuulutus hyväksytty ELY/1/2022",
    "text": "Valtion liikenneväylien suunnittelu -järjestelmän projektin
Testiprojekti 1
aloituskuulutus on hyväksytty. Liitteenä aloituskuulutus PDF-tiedostona, muistathan viedä sen asiakirjanhallintaan.

Voit tarkastella aloituskuulutusta osoitteessa https://localhost:3000/yllapito/projekti/1/aloituskuulutus

Saat tämän viestin, koska sinut on merkitty aloituskuulutuksen projektipäälliköksi. Tämä on automaattinen sähköposti, johon ei voi vastata.",
    "to": Array [
      "pekka.projari@vayla.fi",
    ],
  },
  Object {
    "attachments": Array [
      Object {
        "contentType": "application/pdf",
        "filename": "T412_1 Ilmoitus aloituskuulutuksesta.pdf",
      },
    ],
    "subject": "Nokia kuulutuksesta ilmoittaminen",
    "text": "Testiprojekti 1

Nokia ilmoittaa, että se julkaisee kuulutuksen, joka koskee otsikossa mainitun tiesuunnitelman ja maastotöiden aloittamista. Nokia saattaa asian tiedoksi julkisesti kuuluttamalla vähintään 30 päivän ajan siten, kuin julkisesta kuulutuksesta säädetään hallintolaissa, sekä julkaisemalla kuulutuksen vähintään yhdessä alueella yleisesti ilmestyvässä sanomalehdessä. (LjMTL 108§, HL 62 a §)

Kuulutus julkaistaan 2.1.2022 osoitteessa https://localhost:3000/suunnitelma/1/aloituskuulutus.

Suunnittelualueen kuntia pyydetään julkaisemaan liitteenä oleva ilmoitus kuulutuksesta verkkosivuillaan. Ilmoitus tulee julkaista mahdollisuuksien mukaan edellä mainittuna kuulutuksen julkaisupäivänä. Ilmoituksen nähtävillä oloa ei tarvitse todentaa ilmoituksen lähettäjälle. Ilmoitus menee ELY-keskuksen ja Väyläviraston kuulutussivulle automaattisesti Valtion liikenneväylien suunnittelu palvelun kautta.

Lorem Ipsum

Uudenmaan ELY-keskus käsittelee suunnitelman laatimiseen liittyen tarpeellisia henkilötietoja. Halutessasi tietää tarkemmin väyläsuunnittelun tietosuojakäytänteistä tutustu verkkosivujen tietosuojaosioon, https://www.ely-keskus.fi/tietosuoja.

Lisätietoja antaa 

Nokia, Kunta Kuntalainen, puh. 123456789, Kunta.Kuntalainen@vayla.fi

Pirkanmaan ELY-keskus, Eemil Elylainen, puh. 123456789, eemil.elylainen@ely.fi

Kajaani, Marko Koi, puh. 0293121213, markku.koi@koi.com",
    "to": Array [],
  },
]
`;
